### PR TITLE
[DOM Reference] Document NamedNodeMap and children

### PR DESCRIPTION
**Pull request by teoli2003**
Part of #9740 

- Update the DOM sidebar by removing non-existing interfaces (deleted a long time ago)
-  Created the missing pages of `NamedNodeMap`.
- Updated the 2 pages that were existing.

Note: bcd companion PR browser-compat-data#13450 adds the missing `spec_url` entries.

### CODE REVIEW

File: files/en-us/web/api/namednodemap/getnameditem/index.md [hunk 1]
```diff
@@ -2,27 +2,59 @@
...
 ---
 {{APIRef("DOM")}}
 
-The **`getNamedItem()`** method of the
-{{domxref("NamedNodeMap")}} interface returns the {{domxref("Attr")}} corresponding to
-the given name, or null if there is no corresponding attribute.
+The **`getNamedItem()`** method of the {{domxref("NamedNodeMap")}} interface returns the {{domxref("Attr")}} corresponding to the given name, or `null` if there is no corresponding attribute.
+
+> **Note:** This method is also called when you use the operator `[]` syntax.
+> So, `myMap[str]`is equivalent to `myMap.getNamedItem(str)` when `str` is a string.
```
Comment (jpmedley) [inline files/en-us/web/api/namednodemap/getnameditem/index.md:14]:
```suggestion
> So, `myMap[str]`is equivalent to `myMap.getNamedItem(str)` where `str` is a string.
```

File: files/en-us/web/api/namednodemap/item/index.md [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+  - Reference
+browser-compat: api.NamedNodeMap.item
+---
+{{APIRef("DOM")}}
+
+The **`item()`** method of the {{domxref("NamedNodeMap")}} interface
+returns the item in the map matching the index.
+
+> **Note:** This method is also called when you use the operator `[]` syntax.
+> So, `myMap[i]`is equivalent to `myMap.item(i)` when `i` is a {{jsxref("Number")}}.
```
Comment (jpmedley) [inline files/en-us/web/api/namednodemap/item/index.md:15]:
```suggestion
> So, `myMap[i]`is equivalent to `myMap.item(i)` where `i` is a {{jsxref("Number")}}.
```

File: files/en-us/web/api/namednodemap/item/index.md [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+```
+
+### Parameters
+
+- `index`
+  - : A {{jsxref("Number")}} representing the index of the item you want to return.
+
+### Return value
+
+An {{domxref("Attr")}} or `null`if the number is greater than or equal to the `length` of
```
Comment (jpmedley) [inline files/en-us/web/api/namednodemap/item/index.md:31]:
```suggestion
An {{domxref("Attr")}} or `null` if the number is greater than or equal to the `length` of
```

File: files/en-us/web/api/namednodemap/length/index.md [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+browser-compat: api.NamedNodeMap.length
+---
+{{APIRef("DOM")}}
+
+The read-only **`length`** property of the {{domxref("NamedNodeMap")}} interface
+is the number of objects stored in the map.
+
+## Value
+
+An {{jsxref("Number")}} containing the number of objects in the map
```
Comment (jpmedley) [inline files/en-us/web/api/namednodemap/length/index.md]:
```suggestion
A {{jsxref("Number")}} containing the number of objects in the map
```

File: files/en-us/web/api/namednodemap/removenameditem/index.md [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+## Syntax
+
+```js
+removeNamedItem(attrName);
+```
+
+### Parameters
+
+- `attrName`
+  - : the name of the attribute to remove from the map.
```
Comment (jpmedley) [inline files/en-us/web/api/namednodemap/removenameditem/index.md:23]:
```suggestion
  - : The name of the attribute to remove from the map.
```

File: files/en-us/web/api/namednodemap/removenameditemns/index.md [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+slug: Web/API/NamedNodeMap/removeNamedItemNS
+tags:
+  - Method
+  - Reference
+browser-compat: api.NamedNodeMap.removeNamedItemNS
+---
+{{APIRef("DOM")}}
+
+The **`removedNamedItemNS()`** method of the {{domxref("NamedNodeMap")}} interface
+remove the {{domxref("Attr")}} corresponding to the given namespace and local name from the map.
```
Comment (jpmedley) [inline files/en-us/web/api/namednodemap/removenameditemns/index.md:12]:
```suggestion
removes the {{domxref("Attr")}} corresponding to the given namespace and local name from the map.
```

File: files/en-us/web/api/namednodemap/getnameditem/index.md (hunk 1 above)
Comment (jpmedley) [reply files/en-us/web/api/namednodemap/getnameditem/index.md:14]: I'm not being nit-picky here. This change is consistent with how such things are generally phrased in English.

File: files/en-us/web/api/namednodemap/removenameditemns/index.md [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+## Syntax
+
+```js
+removeNamedItem(namespace, localName);
+```
+
+### Parameters
+
+- `namespace`
+  - : the namespace of the attribute to remove from the map
```
Comment (jpmedley) [inline files/en-us/web/api/namednodemap/removenameditemns/index.md:23]:
```suggestion
  - : The namespace of the attribute to remove from the map
```

File: files/en-us/web/api/namednodemap/removenameditemns/index.md [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+    > **Warning:** `namespace`is the URI of the namespace, not the prefix.
```
Comment (jpmedley) [inline files/en-us/web/api/namednodemap/removenameditemns/index.md:24]:
```suggestion
    > **Warning:** `namespace` is the URI of the namespace, not the prefix.
    
```

Comment (jpmedley) [reply files/en-us/web/api/namednodemap/removenameditemns/index.md:24]: The extra line is not a mistake. Without it, the definition of `localName` is swallowed by the warning.

File: files/en-us/web/api/namednodemap/removenameditemns/index.md [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+- `localName`
+  - : the local name of the attribute to remove from the map.
```
Comment (jpmedley) [inline files/en-us/web/api/namednodemap/removenameditemns/index.md:26]:
```suggestion
  - : The local name of the attribute to remove from the map.
```

File: files/en-us/web/api/namednodemap/setnameditem/index.md [hunk 10]
```diff
@@ -0,0 +1,75 @@
...
+slug: Web/API/NamedNodeMap/setNamedItem
+tags:
+  - Method
+  - Reference
+browser-compat: api.NamedNodeMap.setNamedItem
+---
+{{APIRef("DOM")}}
+
+The **`setNamedItem()`** method of the {{domxref("NamedNodeMap")}} interface
+put the {{domxref("Attr")}} identified by its name in the map.
```
Comment (jpmedley) [inline files/en-us/web/api/namednodemap/setnameditem/index.md:12]:
```suggestion
puts the {{domxref("Attr")}} identified by its name in the map.
```

File: files/en-us/web/api/namednodemap/setnameditem/index.md [hunk 11]
```diff
@@ -0,0 +1,75 @@
...
+If there was already an {{domxref("Attr")}} with the same name in the map,
```
Comment (jpmedley) [inline files/en-us/web/api/namednodemap/setnameditem/index.md:13]:
```suggestion
If there is already an {{domxref("Attr")}} with the same name in the map,
```

File: files/en-us/web/api/namednodemap/setnameditem/index.md [hunk 12]
```diff
@@ -0,0 +1,75 @@
...
+```
+
+### Parameters
+
+- `attr`
+  - : the attribute to insert in the map.
+
+### Return value
+
+Return the old attribute if replaced, or `null` if the attribute is new.
```
Comment (jpmedley) [inline files/en-us/web/api/namednodemap/setnameditem/index.md:29]:
```suggestion
Returns the old attribute if replaced, or `null` if the attribute is new.
```

File: files/en-us/web/api/namednodemap/setnameditemns/index.md [hunk 13]
```diff
@@ -0,0 +1,81 @@
...
+slug: Web/API/NamedNodeMap/setNamedItemNS
+tags:
+  - Method
+  - Reference
+browser-compat: api.NamedNodeMap.setNamedItemNS
+---
+{{APIRef("DOM")}}
+
+The **`setNamedItemNS()`** method of the {{domxref("NamedNodeMap")}} interface
+put the {{domxref("Attr")}} identified by its name in the map.
```
Comment (jpmedley) [inline files/en-us/web/api/namednodemap/setnameditemns/index.md:12]:
```suggestion
puts the {{domxref("Attr")}} identified by its name in the map.
```

File: files/en-us/web/api/namednodemap/setnameditemns/index.md [hunk 14]
```diff
@@ -0,0 +1,81 @@
...
+```
+
+### Parameters
+
+- `attr`
+  - : the attribute to insert in the map.
+
+### Return value
+
+Return the old attribute if replaced, or `null` if the attribute is new.
```
Comment (jpmedley) [inline files/en-us/web/api/namednodemap/setnameditemns/index.md:32]:
```suggestion
Returns the old attribute if replaced, or `null` if the attribute is new.
```

File: files/en-us/web/api/namednodemap/setnameditemns/index.md [hunk 15]
```diff
@@ -0,0 +1,81 @@
...
+## Syntax
+
+```js
+setNamedItemNS(attr);
+```
+
+### Parameters
+
+- `attr`
+  - : the attribute to insert in the map.
```
Comment (jpmedley) [inline files/en-us/web/api/namednodemap/setnameditemns/index.md:28]:
```suggestion
  - : The attribute to insert in the map.
```

File: files/jsondata/GroupData.json [hunk 16]
```diff
@@ -288,9 +288,6 @@
         "ByteString",
```
Comment (jpmedley) [inline files/jsondata/GroupData.json:288]: @ddbeck I confess that I've never completely understood GroupData.json. I can't evaluate what's been done. Can you please review this or pass it to someone who can?

**Review by jpmedley — CHANGES_REQUESTED** [review]

File: files/en-us/web/api/namednodemap/removenameditem/index.md [hunk 17]
```diff
@@ -0,0 +1,67 @@
...
+slug: Web/API/NamedNodeMap/removeNamedItem
+tags:
+  - Method
+  - Reference
+browser-compat: api.NamedNodeMap.removeNamedItem
+---
+{{APIRef("DOM")}}
+
+The **`removeNamedItem()`** method of the {{domxref("NamedNodeMap")}} interface
+remove the {{domxref("Attr")}} corresponding to the given name from the map.
```
Comment (ddbeck) [inline files/en-us/web/api/namednodemap/removenameditem/index.md:12]:
```suggestion
removes the {{domxref("Attr")}} corresponding to the given name from the map.
```

**Review by ddbeck — CHANGES_REQUESTED** [review]
Had a look at the GroupData and reviewed the rest while I was here. Thanks, @teoli2003.

File: files/jsondata/GroupData.json [hunk 18]
```diff
@@ -299,37 +296,30 @@
...
         "ProcessingInstruction",
-        "PromiseResolver",
         "Range",
         "StaticRange",
         "Text",
         "TextDecoder",
         "TextEncoder",
         "TimeRanges",
         "TreeWalker",
-        "USVString",
```
[the remark this answers is not in the document]
Comment (teoli2003) [reply files/jsondata/GroupData.json:332]: Yes, `USVString`must go.

File: files/jsondata/GroupData.json [hunk 19]
```diff
@@ -288,9 +288,6 @@
         "ByteString",
         "CDATASection",
         "CharacterData",
-        "CSSPrimitiveValue",
-        "CSSValue",
-        "CSSValueList",
```
[the remark this answers is not in the document]
Comment (teoli2003) [reply files/jsondata/GroupData.json:293]: They are not part of the DOM but of CSSOM.

File: files/en-us/web/api/namednodemap/getnameditem/index.md (hunk 1 above)
Comment (teoli2003) [reply files/en-us/web/api/namednodemap/getnameditem/index.md]: Thanks, it is one of my "usual errors" that I'm trying to correct.

File: files/jsondata/GroupData.json [hunk 20]
```diff
@@ -299,37 +296,30 @@
         "DOMError",
         "DOMException",
         "DOMImplementation",
-        "DOMObject",
         "DOMParser",
         "DOMPoint",
-        "DOMPointInit",
         "DOMPointReadOnly",
         "DOMRect",
-        "DOMString",
```
[the remark this answers is not in the document]
Comment (teoli2003) [reply files/jsondata/GroupData.json:308]: Yes DOMString, like USVString, will go away when all links to them will be removed. They are invisible to the web devs.

File: files/jsondata/GroupData.json (hunk 18 above)
[the remark this answers is not in the document]
Comment (teoli2003) [reply files/jsondata/GroupData.json:332]: The plan is to say `string` where it is used and to add a sentence saying which restriction it has from another string. There is `ByteString`that is similar too.

File: files/en-us/web/api/namednodemap/length/index.md (hunk 4 above)
Comment (teoli2003) [reply files/en-us/web/api/namednodemap/length/index.md]: Fixed (couldn't directly commit this suggestion as the comment was marked "Outdated")

File: files/en-us/web/api/namednodemap/getnameditemns/index.md [hunk 21]
```diff
@@ -0,0 +1,60 @@
...
+
+```js
+getNamedItemNS(namespace, localName);
+```
+
+### Parameters
+
+- `namespace`
+  - : A string with the namespace URI of the desired attribute.
+    > **Warning:** `namespace`is the URI of the namespace, not the prefix.
```
[the remark this answers is not in the document]
Comment (teoli2003) [reply files/en-us/web/api/namednodemap/getnameditemns/index.md:23]: Looks OK iin preview:
image

File: files/en-us/web/api/namednodemap/removenameditemns/index.md (hunk 8 above)
Comment (teoli2003) [reply files/en-us/web/api/namednodemap/removenameditemns/index.md:24]: I'll commit but I don't think the extra line is needed. Here is the screenshot from the preview:
image

File: files/en-us/web/api/namednodemap/getnameditemns/index.md (hunk 21 above)
Comment (teoli2003) [inline files/en-us/web/api/namednodemap/getnameditemns/index.md:23]:
```suggestion
    > **Warning:** `namespace` is the URI of the namespace, not the prefix.
```

**Review by jpmedley — APPROVED** [review]

File: files/en-us/web/api/namednodemap/getnameditemns/index.md (hunk 21 above)
[the remark this answers is not in the document]
Comment (ddbeck) [reply files/en-us/web/api/namednodemap/getnameditemns/index.md]: Nice. Thank you!

**Review by ddbeck — APPROVED** [review]
Looks great. Thank you! 🎉

**Comment by ddbeck** [comment]
(The failures are an artifact of the introduction of Netlify for contributor docs in December. It's ignorable in this case. Going ahead with a merge anyway.)

**Pull request merged**